api_docs/changelog.md

```diff
@@ -20,6 +20,18 @@ format used by the Zulip server that they are interacting with.
 
 ## Changes in Zulip 7.0
 
+**Feature level 162**
+
+* [`GET /users`](/api/get-users), [`GET /users/{user_id}`](/api/get-user),
+  [`GET /users/{email}`](/api/get-user-by-email),
+  [`GET /users/me`](/api/get-own-user) and [`GET /events`](/api/get-events):
+  `delivery_email` field is passed in user objects always with value being `None`
+  if user does not have access to real email and also when `email_address_visibility`
+  is set to `EMAIL_ADDRESS_VISIBILITY_EVERYONE`.
```

Comment:
Hmm, it seems like the API would be simpler if it was just identical to `email` if `EMAIL_ADDRESS_VISIBILITY_EVERYONE`? I think a correct client could "Display an email for a user if and only if `delivery_email` is not None) -- and never display the fake `email` field except as required for legacy API details like the search box that we're hoping to eliminate.

Reply:
Ohh, the way it is written here is confusing I guess. We set both `email` and `delivery_email` to real email for `EMAIL_ADDRESS_VISIBILITY_EVERYONE`. I wanted to write that previoulsy there was no `delivery_email` field in the object for `EMAIL_ADDRESS_VISIBILITY_EVERYONE`, but now there is. I agree that the way it is phrased makes it confusing.